test/extensions/quic_listeners/quiche/envoy_quic_alarm_test.cc

```diff
@@ -0,0 +1,148 @@
+#include "common/event/libevent_scheduler.h"
+
+#include "extensions/quic_listeners/quiche/envoy_quic_alarm.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_alarm_factory.h"
+#include "extensions/quic_listeners/quiche/platform/envoy_quic_clock.h"
+
+#include "test/test_common/simulated_time_system.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using Envoy::Event::Dispatcher;
+using quic::QuicTime;
+
+namespace Envoy {
+namespace Quic {
+
+class TestDelegate : public quic::QuicAlarm::Delegate {
+public:
+  TestDelegate() : fired_(false) {}
+
+  void OnAlarm() override { fired_ = true; }
+  bool fired() const { return fired_; }
+
+private:
+  bool fired_;
+};
+
+class EnvoyQuicAlarmTest : public ::testing::Test {
+public:
+  EnvoyQuicAlarmTest()
+      : clock_(time_system_), scheduler_(time_system_.createScheduler(base_scheduler_)),
+        alarm_factory_(*scheduler_, clock_) {}
+
+  void advanceUsAndLoop(int64_t delay_us) {
+    time_system_.sleep(std::chrono::microseconds(delay_us));
+    base_scheduler_.run(Dispatcher::RunType::NonBlock);
+  }
+
+protected:
+  Event::SimulatedTimeSystemHelper time_system_;
+  EnvoyQuicClock clock_;
```

Comment:
where does this come from? I am not seeing it in the PR.

Reply:
Added in #7028